language/stdlib/modules/DesignatedDealer.move

```diff
@@ -121,8 +137,28 @@ module DesignatedDealer {
         add_tier<CoinType>(tc_account, dd_addr, TIER_3_DEFAULT * coin_scaling_factor);
     }
     spec fun add_currency {
-        /// >TODO: times out without any specification, perhaps it won't if fully specified.
-        pragma verify_duration_estimate = 100;
+        pragma opaque = true;
+        let dd_addr = Signer::spec_address_of(dd);
+        modifies global<TierInfo<CoinType>>(dd_addr);
+        ensures exists<TierInfo<CoinType>>(dd_addr);
+        ensures global<TierInfo<CoinType>>(dd_addr).window_start == LibraTimestamp::spec_now_microseconds();
+        ensures global<TierInfo<CoinType>>(dd_addr).window_inflow == 0;
+        ensures len(global<TierInfo<CoinType>>(dd_addr).tiers) == 4;
+
+        include Roles::AbortsIfNotTreasuryCompliance{account: tc_account};
```

Comment:
See above about order of aborts_if/ensures.

Reply:
Done.